### ISSUE TITLE
2 Suggestion : need more configurable for Stonecutter and Giant

### ISSUE THREAD

**Issue by cardanawandra** [pl3xgaming/Purpur#318]
please add this feature
Stonecutter
1. enable hurt while shifting
2. specify any mobs who immune to stonecutter

Giant
1. makes special attack ride
2. natural spawn setup
3. better hitboxes

**Comment by BillyGalbreath** [comment]
I can fix the stonecutter not doing damage when sneaking, no problem.

As for specifying mob types that take damage, a tiny plugin can be written to handle that. Its a bit excessive to add that directly to the server. As it stands now, most mobs should be avoiding the stonecutters the same way they avoid other blocks that cause them damage.

The giant stuff is a no go. Giants have never had a special attack, all I did was give them their old AI back. Natural spawns is something Purpur _used_ to handle, but with 1.16.2 they changed a lot of that code and its now handled using datapacks. The hitboxes is something I can't fix, as that is is also coded client side (even if I make the hitbox bigger on server, the client will still miss so wont send the server any interact packets).

**Comment by LaurenceBarnes** [comment]
If you want Giants, this might do what you want

https://www.spigotmc.org/resources/better-giants.76505/

**Comment by BillyGalbreath** [comment]
> If you want Giants, this might do what you want
> 
> https://www.spigotmc.org/resources/better-giants.76505/

That plugin will replace a few zombies with custom giants. Those giants will not use any of the giant settings in purpur.yml

Like I mentioned before, just use create a datapack to add giants to your game. I even posted a base example datapack in discord to help get you started. Just download the pack, open it up in 7zip and edit it to the way you want ^_^

https://discord.com/channels/685683385313919172/763081571363323914/763100407525933087

**Comment by LaurenceBarnes** [comment]
I know, but good thing about this plugin is that the giants have some cool attacks :p
Plus it also checks spawn conditions (space mostly), the giants have a unique hitboxes and configurable item drops (to a certain extent)

**Comment by cardanawandra** [comment]
NICE!, very very thank you, but are there any anti stonecutter damage plugins?

or even just a code, i will compile it

**Comment by LaurenceBarnes** [comment]
Turn off the stonecutter damage?

**Comment by TetraTheta** [comment]
> specify any mobs who immune to stonecutter

I think that can be done by checking these:

1. DamageCause(DamageSource?) is CACTUS
2. LivingEntity is on stonecutter
3. LivingEntity is listed on the 'immune' list

then canceling the event will prevent them to get damage.

Maybe i can make plugin for this :)

**Comment by cardanawandra** [comment]
@TetraTheta  WHOA AWESOME DUDES

also please add Stonecutter sit functionality when right click bare hand + shifting

**Comment by TetraTheta** [comment]
> also please add Stonecutter sit functionality when right click bare hand + shifting

Unfortunately, for that request, the answer is no.  
The reason I've made the plugin is those functionality should be done in plugin-scope, not server-scope. Imagine `purpur.yml` is flooded by entitiy list of 'Stonecutter immune' stuff.  
And adding 'sit' functionality is far from what described first, 'specify any mobs who immune to stonecutter'. So I regard that as another 'request' which I am not interested in.

If you want to add that functionality, you would have to find someone else to do that for you.

Anyway, I've already made a plugin about stonecutter immune stuff. I think if I post direct repository link, it would be treated as advertisement. So I will just give you a hint. _Look for something related to it in Contribution activity in my profile._
